contracts/staking/ServiceStakingBase.sol

```diff
@@ -202,96 +291,115 @@ abstract contract ServiceStakingBase is IErrorsRegistries {
 
-        // If available rewards are not zero, proceed with staking calculation
-        if (lastAvailableRewards > 0) {
-            // Get necessary arrays
-            eligibleServiceIds = new uint256[](size);
-            eligibleServiceRewards = new uint256[](size);
-
-            // Calculate each staked service reward eligibility
-            for (uint256 i = 0; i < size; ++i) {
-                // Get the service info
-                uint256 curServiceId = serviceIds[i];
-                ServiceInfo storage curInfo = mapServiceInfo[curServiceId];
-
-                // Calculate the liveness nonce ratio
-                // Get the last service checkpoint: staking start time or the global checkpoint timestamp
-                uint256 serviceCheckpoint = tsCheckpointLast;
-                // Adjust the service checkpoint time if the service was staking less than the current staking period
-                if (curInfo.tsStart > tsCheckpointLast) {
-                    serviceCheckpoint = curInfo.tsStart;
-                }
-                // Calculate the liveness ratio in 1e18 value
-                uint256 ratio;
-                // If the checkpoint was called in the exactly same block, the ratio is zero
-                if (block.timestamp > serviceCheckpoint) {
-                    uint256 nonce = serviceNonces[i];
-                    ratio = ((nonce - curInfo.nonce) * 1e18) / (block.timestamp - serviceCheckpoint);
-                }
+        // Check the last checkpoint timestamp and the liveness period
+        uint256 tsCheckpointLast = tsCheckpoint;
+        if (block.timestamp - tsCheckpointLast >= livenessPeriod) {
+            // Get available rewards and last checkpoint timestamp
+            lastAvailableRewards = availableRewards;
+
+            // If available rewards are not zero, proceed with staking calculation
+            if (lastAvailableRewards > 0) {
+                // Get necessary arrays
+                eligibleServiceIds = new uint256[](size);
+                eligibleServiceRewards = new uint256[](size);
+                serviceNonces = new uint256[](size);
+
+                // Calculate each staked service reward eligibility
+                for (uint256 i = 0; i < size; ++i) {
+                    // Get the service info
+                    ServiceInfo storage curInfo = mapServiceInfo[serviceIds[i]];
+
+                    // Get current service multisig nonce
+                    serviceNonces[i] = IMultisig(curInfo.multisig).nonce();
+
+                    // Calculate the liveness nonce ratio
+                    // Get the last service checkpoint: staking start time or the global checkpoint timestamp
+                    uint256 serviceCheckpoint = tsCheckpointLast;
+                    uint256 tsStart = curInfo.tsStart;
+                    // Adjust the service checkpoint time if the service was staking less than the current staking period
+                    if (tsStart > serviceCheckpoint) {
```

Comment:
With this approach, we are not forcing a service to be staked for the entire liveness period. As we did previously, service multisigs that pass the liveness check will be considered for incentives, even if they have been staked for less than the liveness period. Is that correct?

Reply:
That is correct, as the checkpoint is a centralization point and everybody eligible must account for their rewards.